### DISCUSSION TITLE
Remote updating wlan.ini (solution)

### DISCUSSION THREAD

**Discussion by Terr4** [jomjol/AI-on-the-edge-device#1389] [Show and tell]
Hi guys,

So I had the issue that I needed to change the SSID remotely with no access to the SD-card but as mentioned several times in the forum this seems to be not possible.

Inventive how I am I made it happen. Not sure if this is considered a bug but at least you cannot read the existing file with the password.
I did this now 2 times already so it definitely works.

You cannot see or read the wlan.ini through the fileserver UI, also you cannot overwrite it with the fileserver UI, it will tell you a file with the same name already exist when you try to upload such a file.

So the trick is that you first need to delete the existing wlan.ini file and then upload a new file through the fileserver UI with the new WiFi settings.

1. Prepare your new wlan.ini file, open Filseserver UI in root and select the File to upload (do not press upload yet)
2. Run the following POST request, e.g. with CURL, powershell, etc. from your PC:
`curl "http://<watermeterIP>/delete//wlan.ini" -X "POST" -H "Content-Length: 0" -H "Content-Type: application/x-www-form-urlencoded"`
You should get the return "File successfully deleted"
3. Upload your new wlan.ini file through Fileserver UI that you prepared, it will succeed this time
4. Reboot through UI and pray that you entered the right SSID and credentials in the file

**Comment by caco3** [comment]
thanks for sharing.
I would see this as a bug.
It is not very critical since the web UI is not secured anyway.

Another easy way would be to upload a (temporary) firmware which has no restriction for that file.

**Reply by caco3** [reply]
https://github.com/jomjol/AI-on-the-edge-device/issues/1390
